### ISSUE TITLE
Date validation doesn't work

### ISSUE THREAD

**Issue by rodic** [parris/iz#37]
```js
var rules = {
  "myDate": [
    {
      "rule": "blank",
      "error": "Additional insured cannot be blank"
    },
    {
      "rule": "date",
      "error": "Additional insured must be valid date"
    },
  ],
};

const result = are(rules).for({
  myDate: "09/23/2012"
});
```
this throws an exception
```
/home/aleksandar/Code/theguarantors.com/node_modules/iz/lib/utils/nameOfClass.js:6
  if (obj.constructor && obj.constructor.displayName) {
         ^
TypeError: Cannot read property 'constructor' of undefined
    at izGetObjectClass (/home/aleksandar/Code/theguarantors.com/node_modules/iz/lib/utils/nam
eOfClass.js:6:10)                                        
```
param to `izDate` is `undefined`.

Apologies if I misunderstood something, but didn't have issues with other validators.

**Comment by parris** [comment]
Huh! I'll take a look. Thanks for reporting.

**Comment by parris** [comment]
@rodic this should be fixed now in - v2.1.4 - you caught a more serious issue actually. Anyways, I created a unit test to capture this issue. 

https://github.com/parris/iz/blob/master/spec/are.spec.js#L128-L143

**Comment by rodic** [comment]
@parris thanks for fixing this.
